kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/catalogManaged/CatalogManagedE2EReadSuite.scala

```diff
@@ -293,4 +299,102 @@ class CatalogManagedE2EReadSuite extends AnyFunSuite with TestUtilsWithTableMana
         expectedFileList)
     }
   }
+
+  // We test this in the unit tests as well, but since those use the withProtocolAndMetadata API
+  // we also test it here with a real table where we load the P&M from the log
+  test("reading a catalogManaged table without providing maxCatalogVersion fails") {
+    withCatalogOwnedPreviewTestTable { (tablePath, parsedLogData) =>
+      // With logData
+      intercept[IllegalArgumentException] {
+        TableManager
+          .loadSnapshot(tablePath)
+          .withLogData(parsedLogData.asJava)
+          .build(defaultEngine)
+      }
+      // Without logData (and with time-travel-version)
+      intercept[IllegalArgumentException] {
+        TableManager
+          .loadSnapshot(tablePath)
+          .atVersion(0)
+          .build(defaultEngine)
+      }
+    }
+  }
+
+  test("reading a file-system managed table and providing maxCatalogVersion fails") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      // Create a basic file-system managed table
+      spark.range(10).write.format("delta").save(tablePath)
```

Comment:
could we use kernel instead to create + write? that would be faster

Reply:
This is just less lines of code since we don't have our write utils here. Happy to do differently.

Reply:
Yeah -- let's use Kernel 👍